apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/*global Compose, Recipients, Utils, Template, Settings,
+         SMIL, ErrorDialog, MessageManager, LinkHelper,
+         ActivityPicker, InboxView, OptionMenu, Threads, Contacts,
+         Attachment, WaitingScreen, MozActivity, LinkActionHandler,
+         TimeHeaders, ContactRenderer, Draft, Drafts,
+         MultiSimActionButton, Navigation, Promise, LazyLoader,
+         App,
+         SharedComponents,
+         ActivityClient,
+         Errors,
+         EventDispatcher,
+         SelectionHandler,
+         TaskRunner
+*/
+/*exported ConversationView */
+
+(function(exports) {
+'use strict';
+
+var attachmentMap = new WeakMap();
+
+function conv_mmsAttachmentClick(target) {
+  var attachment = attachmentMap.get(target);
+  if (!attachment) {
+    return false;
+  }
+
+  attachment.view({
+    allowSave: true
+  });
+
+  return true;
+}
+
+// reduce the Composer.getContent() into slide format used by SMIL.generate some
+// day in the future, we should make the SMIL and Compose use the same format
+function conv_generateSmilSlides(slides, content) {
+  var length = slides.length;
+  if (typeof content === 'string') {
+    if (!length || slides[length - 1].text) {
+      slides.push({
+        text: content
+      });
+    } else {
+      slides[length - 1].text = content;
+    }
+  } else {
+    slides.push({
+      blob: content.blob,
+      name: content.name
+    });
+  }
+  return slides;
+}
+
+var ConversationView = {
+  CHUNK_SIZE: 10,
+  // duration of the notification that message type was converted
+  CONVERTED_MESSAGE_DURATION: 3000,
+  IMAGE_RESIZE_DURATION: 3000,
+  BANNER_DURATION: 2000,
+
+  // Toast duration when you write a long text and need more than one SMS
+  // to send it
+  ANOTHER_SMS_TOAST_DURATION: 3000,
+
+  // when sending an sms to several recipients in activity, we'll exit the
+  // activity after this delay after moving to the thread list.
+  LEAVE_ACTIVITY_DELAY: 3000,
+
+  draft: null,
+  recipients: null,
```

Comment:
> Not quite understand this part: In the ConversationView we would only need an array of recipient and it will contain in thread object. But the "recipients" here means the recipients module, and we don't really need most of the functionalities in conversation view.

Yeah, I think the same... @julienw for what case/reason you wanted to have a copy of ```ConversationView.conversation.participants``` in  ```ConversationView.recipients```? I think that could be useful for non split views, when we save draft or send message so that we can avoid different ```if```, but not sure if it makes sense for split views?